### PR TITLE
chore(python): clean up bytecode parsing a bit

### PR DESCRIPTION
**Pull request by MarcoGorelli**
There's 3 cleanups I'm suggesting here:

1. make `try` blocks smaller. I think this is uncontroversial
2. remove an unused `try` block in `_is_raw_function`. The whole test suite passes without it, I think it the code can be simplified by just removing it
3. removing the `_can_attempt_rewrite` dict. I think this was originally introduced for performance reasons, but:
   - I'm not seeing any perf difference here vs on `main` (timing `pl.col('a').map_elements(lambda x: abs(x))` here and on `main` takes about `8.988900013459247e-05` seconds on my laptop
   - it adds quite a lot of complexity
   
   It's already super-fast, I don't think the extra complexity is warranted

### CODE REVIEW

File: py-polars/polars/utils/udfs.py
Comment (MarcoGorelli) [inline py-polars/polars/utils/udfs.py:339]: this can all be simplified by removing the `_can_attempt_rewrite` dict

Comment (MarcoGorelli) [reply py-polars/polars/utils/udfs.py:339]: the `BytecodeParser` class needs initialising once for each call anyway, so I'm not sure what `_can_attempt_` is supposed to be caching

**Comment by MarcoGorelli** [comment]
@alexander-beedie fancy taking a look?

**Comment by alexander-beedie** [comment]
> @alexander-beedie fancy taking a look?

Definitely! Can dig in tonight (or tomorrow morning) and see if I can recall the primary scenario this speeds up (or used to speed up 😂)

File: py-polars/polars/utils/udfs.py
Comment (alexander-beedie) [reply py-polars/polars/utils/udfs.py:339]: Think I've got it; this used to accelerate things when the code was more self-referential, before I split it out into more specialised classes. Seems like this came along for the ride, but I think you are right and it is no longer necessary! I say clean it up 🤔

Comment (MarcoGorelli) [reply py-polars/polars/utils/udfs.py:339]: thanks! 

 > I think you are right and it is no longer necessary!

gentle reminder that I can't merge without approval (in case this was meant as implicit approval, no hurry though!)

Comment (alexander-beedie) [reply py-polars/polars/utils/udfs.py:339]: For some reason thought there was more to come, oops!
Hitting the big green button now 😅

**Review by alexander-beedie — APPROVED** [review]

**Pull request merged**
